### DISCUSSION TITLE
ambient-mode in a multi-tenant cluster - How does mTLS key management work here?

### DISCUSSION THREAD

**Discussion by m8-t** [istio/istio#52894] [Q&A] [answered]
Hi,
we are close to deploy istio into one of our clusters to OpenShift. Thus ambient-mode is officially not supported yet we will deploy istio in ambient-mode. One question raised in my head regarding mTLS in ambient-mode. [This](https://istio.io/latest/docs/overview/dataplane-modes/#choosing-between-sidecar-and-ambient) page states under "Security model" for the ambient mode "Strong: each node agent has only the keys for workloads on that node" versus "Strongest: each workload has its own keys" for sidecar-mode. Does that mean ambient mode uses  "per-node" keys for any workload on that specific node? For example: We got two customers on a shared node, as far as I understand any traffic is routed through ztunnel, even on the same node. How are these workloads isolated besides Kubernetes namespace/network isolation? Is the traffic encrypted with the same keys even if these projects are unrelated to each other?

Thanks

**Comment by howardjohn** [comment, marked as answer]
https://github.com/istio/istio/blob/3aa1a0d874d0b745371e6056dfc9f064c39646af/architecture/ambient/ztunnel.md#L1 has maybe a bit more info as does https://youtu.be/QnfrbbY_Hy4. I am working on a proper istio.io doc for this but haven't finished yet.

So, each ztunnel will get a certificate for each unique identity (defined by NS+SA) on its node. So if I have 2 pods with `foobar` SA, they will share 1 cert. The CA will enforce that ztunnel never requests certificates for identities not running on its node (so a malicious node cannot compromise the entire cluster).

If you have 2 tenants on the same node, presumably they are from different NS, so they would have different keys.

**Comment by m8-t** [comment]
Thanks! This really cleared up things and I guess we will give ambient a shot with a small user base first.
